File: javascript/grains/grains.js
function Grains() {
  'use strict';

  this.square = function(num) {
    return Math.pow(2, num - 1);
  }

  this.total = function() {
    var result = 0;
    for (var i = 0; i <= 64; i++) {
      result += this.square(i);
    }
    return result;
  }

}

module.exports = Grains

File: go/grains/grains.go
package grains

import "math"

// Square returns the number of grains on the provided square.
func Square(n int) (uint64, error) {
	return uint64(math.Pow(2, float64(n-1))), nil
}

// Total returns the sum of all grains on the chess board.
func Total() uint64 {
	var sum uint64 = 0
	for i := 1; i <= 64; i++ {
		var result, _ = Square(i)
		sum += result
	}
	return sum
}
